### DISCUSSION TITLE
Is it necessary to escape slash (`/`)?

### DISCUSSION THREAD

**Discussion by eostrom-livefront** [i18next/i18next#2199] [General]
We were surprised to find the value "$10/month" interpolated into a `Trans` component as `$1&#x2F;month`. Is it necessary? Resources like [this OWASP cheat sheet](https://cheatsheetseries.owasp.org/cheatsheets/Cross_Site_Scripting_Prevention_Cheat_Sheet.html) and [this Stack Overflow answer](https://stackoverflow.com/questions/7381974/which-characters-need-to-be-escaped-in-html) seem to recommend escaping `&<>"'` but not `/`.

It doesn't matter in the browser, but it was a bit of a hiccup because our unit tests expected to see "$10/month" in the output and instead had to look for the HTML entity. I guess I'm just wondering why.

**Comment by adrai** [comment]
Can you provide a minimal reproducible example of what you exactly mean?
Is it more related to i18next? => https://github.com/i18next/i18next/blob/master/src/utils.js#L151
Or to react-i18next? => https://github.com/i18next/react-i18next/blob/master/src/unescape.js

**Comment by eostrom-livefront** [comment]
Great question, thank you. We encountered it working with `next-i18next` via `react-i18next`'s `Trans` component, but I believe the issue is in the `i18next` `escape` function you linked to.

I'm having some trouble creating a minimal test case. I'll keep working on it, but in the meantime I can point at a test in this repo: https://github.com/i18next/i18next/blob/c8d9f7bbc77ccb1bc1f24272827afd9295281756/test/runtime/interpolation.test.js#L294C1-L298C9

```js
    const tests = [
      {
        args: ['test {{test}}', { test: '<a>foo</a>' }],
        expected: 'test &lt;a&gt;foo&lt;&#x2F;a&gt;',
      },
```

What I'm suggesting is that the slash doesn't need to be escaped, and the expected value should be:

```js
        expected: 'test &lt;a&gt;foo&lt;/a&gt;',
```

**Reply by adrai** [reply]
ufff... that test is there since v1.... (@jamuhl any idea?)

**Reply by jamuhl** [reply]
not sure it's necessary...but was ok for over 10 years 

guess there is a option to disable encoding -> just make sure user inputs get encoded somewhat to avoid suprises ;)

**Comment by adrai** [comment]
Found the original commit that introduced that escape function: https://github.com/i18next/i18next/commit/01082e044d8f0e2fecec702020b7def0d5b2fb96#diff-c761a09fd4bad3ad646a9cfa821ef6010082320baf15c59f3c03daf9f40dd404R59
11 years ago, committed by @anderslemke

**Reply by adrai** [reply]
and here the corresponding issue/PR: https://github.com/i18next/i18next/pull/106
